src/components/sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect, useRef } from "react";
+import { Link } from "react-router-dom";
+import {
+  HistoryIcon,
+  HomeIcon,
+  LikeIcon,
+  PlaylistIcon,
+  WatchLaterIcon,
+} from "../../assets";
+import "./sidebar.css";
+import { useWindowResize } from "beautiful-react-hooks";
+import { useGlobalEvent, useDebouncedCallback } from "beautiful-react-hooks";
+
+const Sidebar = () => {
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+  const onWindowResize = useGlobalEvent("resize");
+
+  const onWindowResizeHandler = useDebouncedCallback(() => {
+    setWindowWidth(window.innerWidth);
+  });
+  onWindowResize(onWindowResizeHandler);
+
+  return (
+    <div
+      className={`sidebar-container flex-column ${
+        windowWidth < 1240 ? "collapsed" : ""
+      }`}
+    >
```

Comment:
appreciation - good that you're checking the window width & applying class to the sidebar